### DISCUSSION TITLE
Cannot use default with `ON UPDATE UNIX_TIMESTAMP()`

### DISCUSSION THREAD

**Discussion by jsardev** [planetscale/discussion#211] [Q&A] [answered]
Hey! So I wanted to create a column in my table that will automatically update on row update using `ON UPDATE`.
```sql
ALTER TABLE `User` MODIFY `updatedAt` BIGINT NOT NULL DEFAULT (UNIX_TIMESTAMP()) ON UPDATE (UNIX_TIMESTAMP());
```
But this does not work and throws an error:
```
target: puser.-.primary: vttablet: rpc error: code = InvalidArgument desc = You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '(UNIX_TIMESTAMP())' at line 1 (errno 1064) (sqlstate 42000) (CallerID: planetscale-admin): Sql: "ALTER TABLE `User` MODIFY `updatedAt` BIGINT NOT NULL DEFAULT (UNIX_TIMESTAMP()) ON UPDATE (UNIX_TIMESTAMP())", BindVars: {}
```
What am I missing? Are actions like `ON UPDATE` etc. not supported by Vitess and PlanetScale?

**Comment by jgreet** [comment, marked as answer]
You can use `ON UPDATE`, but I believe that DDL will throw the same error in plain MySQL 8.

What MySQL intends for you to do is use a timestamp or datetime column and `now()` or one of its many synonyms instead of `unix_timestamp()`.

See https://dev.mysql.com/doc/refman/8.0/en/timestamp-initialization.html

```sql
time/test> CREATE TABLE User (
    ->   id int,
    ->   updatedAt TIMESTAMP NOT NULL DEFAULT NOW() ON UPDATE NOW()
    -> );
time/test> insert into User (id) values (1);
time/test> select * from User\G
*************************** 1. row ***************************
       id: 1
updatedAt: 2022-06-06 17:29:21
```

Then use `unix_timestamp()` if you need the timestamp integer:
```sql
time/test> select id, unix_timestamp(updatedAt) as unix_time from User\G
*************************** 1. row ***************************
       id: 1
unix_time: 1654536561
```

**Reply by jsardev** [reply]
@jgreet Using `DateTime` or `Timestamp` when you are going to work with different timezones are [unreliable and cause a lot of issues](https://stackoverflow.com/a/59460086/4518022). That's why I wanted to use `bigint` and `unix_timestamp()` for that.

I don't know why I assumed this should work. I guess I'll need to handle that on the app side. Sorry for the confusion 😅 
And thank you for your answer! 🙇‍♂️
